File: datascience/lab1/case_studies/case5.R
fmt_money <- function(x) formatC(x, format = "f", digits = 2, big.mark = ",")

for (i in 1:5) {
  cat(sprintf("\n--- Employee %d ---\n", i))

  raw <- readline(sprintf("Name (default Employee %d): ", i))
  name <- if (nchar(raw) == 0) paste("Employee", i) else raw

  raw <- readline("Basic Salary (default 50000): ")
  basic <- if (nchar(raw) == 0) 50000 else as.numeric(raw)

  raw <- readline("Years of Experience (default 3): ")
  years <- if (nchar(raw) == 0) 3 else as.numeric(raw)

  if (years < 2) {
    allowance_rate <- 0.05
  } else if (years <= 5) {
    allowance_rate <- 0.10
  } else {
    allowance_rate <- 0.15
  }

  if (basic > 100000) {
    bonus_rate <- 0.10
  } else {
    bonus_rate <- 0.05
  }

  allowance <- basic * allowance_rate
  bonus <- basic * bonus_rate
  gross <- basic + allowance + bonus

  cat(sprintf("Employee Name: %s\n", name))
  cat(sprintf("Basic Salary: Rs. %s\n", fmt_money(basic)))
  cat(sprintf("Experience: %g year(s)\n", years))
  cat(sprintf("Allowance (%.0f%%): Rs. %s\n", allowance_rate * 100, fmt_money(allowance)))
  cat(sprintf("Bonus (%.0f%%): Rs. %s\n", bonus_rate * 100, fmt_money(bonus)))
  cat(sprintf("Gross Salary: Rs. %s\n", fmt_money(gross)))
}

File: datascience/lab1/py/case_studies/case5.py
for i in range(1, 6):
    print(f"\n--- Employee {i} ---")
    name = input(f"Name (default Employee {i}): ").strip() or f"Employee {i}"
    basic = float(input("Basic Salary (default 50000): ").strip() or 50000)
    years = int(input("Years of Experience (default 3): ").strip() or 3)

    if years < 2:
        allowance_rate = 0.05
    elif years <= 5:
        allowance_rate = 0.10
    else:
        allowance_rate = 0.15

    if basic > 100000:
        bonus_rate = 0.10
    else:
        bonus_rate = 0.05

    allowance = basic * allowance_rate
    bonus = basic * bonus_rate
    gross = basic + allowance + bonus

    print(f"Employee Name: {name}")
    print(f"Basic Salary: Rs. {basic:,.2f}")
    print(f"Experience: {years} year(s)")
    print(f"Allowance ({allowance_rate * 100:.0f}%): Rs. {allowance:,.2f}")
    print(f"Bonus ({bonus_rate * 100:.0f}%): Rs. {bonus:,.2f}")
    print(f"Gross Salary: Rs. {gross:,.2f}")
